hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityConverter.java

```diff
@@ -118,7 +143,34 @@ private BaselineEntity getBaselineEntity(
                 attrMap,
                 containerMap,
                 incomingEdges,
-                outgoingEdges));
+                outgoingEdges,
+                labelResultSet));
+  }
+
+  private LabelResultSet buildLabelResultSet(
+      org.hypertrace.gateway.service.v1.entity.Entity entity,
+      EntityRequest request,
+      Map<String, String> labelsMap) {
+    if (request.labelRequest().isEmpty()) {
+      return ConvertedLabelResultSet.EMPTY_LABEL_RESULT_SET;
+    }
+    Value value =
+        entity.getAttributeOrDefault(
+            request.labelRequest().get().attributeRequest().attribute().id(), null);
+    if (value == null) {
+      return ConvertedLabelResultSet.EMPTY_LABEL_RESULT_SET;
+    }
+    ProtocolStringList labelIds = value.getStringArrayList();
+    if (labelIds.isEmpty() || labelsMap.isEmpty()) {
+      return ConvertedLabelResultSet.EMPTY_LABEL_RESULT_SET;
+    }
+    List<Label> convertedLabels =
+        labelsMap.entrySet().stream()
+            .filter(entry -> labelIds.contains(entry.getKey()))
```

Comment:
we should be going in the other direction - if an entity has a label id that didn't come back in the fetch, we're silently ignoring it here. We probably don't want to error out the response, but we should be logging an error.